servers/src/main/java/tachyon/worker/block/allocator/RRAllocator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.allocator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+/**
+ * A Round robin allocator that allocates a block in the storage dir.
+ * It will allocate the block in the highest tier possible:
+ * It always starts from the highest tier in a RR manner and goes to the next tier when there
+ * is no enough space. 
+ * The allocator only considers non-specific writes in its RR policy (the location is either 
+ * AnyTier or AnyDirInTier).
+ */
+public class RRAllocator implements Allocator {
```

Comment:
for sure. I just followed the old name in 0.6.4

On 2 July 2015 at 17:00, Bin Fan <notifications@github.com> wrote:

> In servers/src/main/java/tachyon/worker/block/allocator/RRAllocator.java
> <https://github.com/amplab/tachyon/pull/1030#discussion_r33833437>:
>
> > +
> > +import tachyon.worker.block.BlockStoreLocation;
> > +import tachyon.worker.block.BlockMetadataManager;
> > +import tachyon.worker.block.meta.StorageDir;
> > +import tachyon.worker.block.meta.StorageTier;
> > +import tachyon.worker.block.meta.TempBlockMeta;
> > +
> > +/**
> > + * A Round robin allocator that allocates a block in the storage dir.
> > + * It will allocate the block in the highest tier possible:
> > + * It always starts from the highest tier in a RR manner and goes to the next tier when there
> > + * is no enough space.
> > + * The allocator only considers non-specific writes in its RR policy (the location is either
> > + * AnyTier or AnyDirInTier).
> > + */
> > +public class RRAllocator implements Allocator {
>
> Can we name this to RoundRobinAllocator instead of RRAllocator
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amplab/tachyon/pull/1030/files#r33833437>.
>



-- 
Tianyin XU,
http://cseweb.ucsd.edu/~tixu/